src/core/net/netif.cpp

```diff
@@ -305,6 +306,19 @@ Error Netif::SubscribeExternalMulticast(const Address &aAddress)
     {
         VerifyOrExit(cur->GetAddress() != aAddress, error = kErrorRejected);
     }
+    // Check that the address is not a solicited node multicast
+    for (const UnicastAddress *cur = mUnicastAddresses.GetHead(); cur; cur = cur->GetNext())
+    {
+        isUnicastBasedAddress = ((aAddress.GetBytes()[8] == cur->GetAddress().GetBytes()[12] &&
+                                  aAddress.GetBytes()[9] == cur->GetAddress().GetBytes()[13] &&
+                                  aAddress.GetBytes()[10] == cur->GetAddress().GetBytes()[14] &&
+                                  aAddress.GetBytes()[11] == cur->GetAddress().GetBytes()[15]) ||
+                                 (aAddress.GetBytes()[8] == cur->GetAddress().GetBytes()[0] &&
+                                  aAddress.GetBytes()[9] == cur->GetAddress().GetBytes()[1] &&
+                                  aAddress.GetBytes()[10] == cur->GetAddress().GetBytes()[2] &&
+                                  aAddress.GetBytes()[11] == cur->GetAddress().GetBytes()[3]));
```

Comment:
Can you please share some background on this change?

Regarding the code, it may be better to wrap this byte-level comparison of an IPv6 address in a method within the `Ip6::Address` class.